clang/lib/CIR/CodeGen/CIRGenBuilder.h

```diff
@@ -408,18 +408,28 @@ class CIRGenBuilderTy : public mlir::OpBuilder {
   /// Get a CIR named struct type.
   mlir::cir::StructType getStructTy(llvm::ArrayRef<mlir::Type> members,
                                     llvm::StringRef name, bool body,
-                                    bool packed, const clang::RecordDecl *ast) {
+                                    bool packed, mlir::Attribute astAttr) {
     const auto nameAttr = getStringAttr(name);
-    std::optional<mlir::cir::ASTRecordDeclAttr> astAttr = std::nullopt;
     auto kind = mlir::cir::StructType::RecordKind::Struct;
-    if (ast) {
-      astAttr = getAttr<mlir::cir::ASTRecordDeclAttr>(ast);
-      kind = getRecordKind(ast->getTagKind());
+    if (astAttr) {
+      if (auto tagDecl = mlir::dyn_cast< mlir::cir::ASTTagDeclInterface >(astAttr)) {
+        kind = getRecordKind(tagDecl.getTagKind());
+      }
     }
     return mlir::cir::StructType::get(getContext(), members, nameAttr, body,
                                       packed, kind, astAttr);
   }
 
+  mlir::cir::StructType getStructTy(llvm::ArrayRef<mlir::Type> members,
+                                    llvm::StringRef name, bool body,
+                                    bool packed, const clang::RecordDecl *ast) {
+    mlir::Attribute astAttr;
+    if (ast) {
```

Comment:
No curly braces needed here.

Reply:
fixed in a6c1d8c990aa849c0a88db914e1af2c858259ae0